pyqtgraph/graphicsItems/LegendItem.py

```diff
@@ -209,13 +214,15 @@ def _addItemToLayout(self, sample, label):
                 row += 1
         self.layout.addItem(sample, row, col)
         self.layout.addItem(label, row, col + 1)
+        # Keep rowCount in sync with the number of rows if items are added
+        self.rowCount = max(self.rowCount, row + 1)
 
     def setColumnCount(self, columnCount):
         """change the orientation of all items of the legend
         """
         if columnCount != self.columnCount:
             self.columnCount = columnCount
-            self.rowCount = int(len(self.items) / columnCount)
+            self.rowCount = math.ceil(len(self.items) / columnCount)
```

Comment:
this will return a float in python2, so we should either cast as an `int()` or do `len(self.items) // columnCount + len(self.items) % columnCount`

Reply:
Good catch on the py2 difference. I'll cast to that. makes the logic clearer.